include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -91,21 +135,29 @@ class fpga_policy : public device_policy<KernelName>
   public:
     static constexpr unsigned int unroll_factor = factor;
 
-    fpga_policy()
-        : base(sycl::queue(
-#    if _ONEDPL_FPGA_EMU
-              __dpl_sycl::__fpga_emulator_selector()
-#    else
-              __dpl_sycl::__fpga_selector()
-#    endif // _ONEDPL_FPGA_EMU
-              ))
-    {
-    }
-
+    fpga_policy() = default;
     template <unsigned int other_factor, typename OtherName>
     fpga_policy(const fpga_policy<other_factor, OtherName>& other) : base(other.queue()){};
```

Comment:
Lazy initialization concept would be broken at this place: ```queue()``` call will create ```sycl::queue``` instance

Reply:
May be we should pass ```static_cast<const base&>(other)``` into constructor of base class to avoid this broken case?

Reply:
Also looks like move constructor of this format and copy/move assignment constructors required in this class too.

Reply:
> Lazy initialization concept would be broken at this place: `queue()` call will create `sycl::queue` instance

That's exactly the question what behavior we want for non-default constructors and assignments. I think I want them being initialized (I can change my mind, though), but I also can see the issues with that since we have inheritance